### ISSUE TITLE
How to destroy chart instance correctly?

### ISSUE THREAD

**Issue by vanchelo** [ChartIQ/Charting-Library---Angular-1.5-Seed-Project#2]
Hi!

How to totally destroy chart instance?
Method `destroy()` does not work correctly.
On resize event listener still present, and i have this error in console:
```
warning: zero width chart. Check CSS for chart container.
bad css style for class stx_yaxis
```
Create chart:
```
const stxx = new CIQ.ChartEngine({
  container: container,
  layout: {
    candleWidth: 16,
    crosshair: true,
    chartType: 'bar'
  }
});

stxx.setTimeZone('America/New_York');
stxx.newChart('SPY', sampleData);
```

Thanks!

**Comment by thorsent** [comment]
@vanchelo You'll need to call stxx.destroy().

**Comment by vanchelo** [comment]
@thorsent 
```
ChartController.prototype.$onDestroy = function () {
  console.info('[CHART] destroy');

  this.stxx.destroy();
};
```

**Comment by vanchelo** [comment]
Full version
```js
export default ChartController;

/*@ngInject*/
function ChartController($element) {
  this.$postLink = function () {
    const container = $element[0].querySelector('[data-chart-container]');
    const stxx = new CIQ.ChartEngine({
      container,
      layout: {
        candleWidth: 16,
        crosshair: true,
        chartType: 'bar'
      }
    });

    stxx.setTimeZone('America/New_York');
    stxx.newChart('SPY', require('./sample-data'));

    this.stxx = stxx;
  };
}

ChartController.prototype.$onDestroy = function () {
  console.info('[CHART] destroy');

  this.stxx.destroy();
};
```

I see chart and after go to another state i see `[CHART] destroy` in console.

**Comment by jaythomas** [comment]
Hi @vanchelo. It appears you have found a bug with the resize event listener. I'll look into this one and let you know when I have a solution. Thanks for reporting it!

**Comment by vanchelo** [comment]
@jaythomas thanks!

**Comment by jaythomas** [comment]
@vanchelo, a fix is a available in the latest version of the ChartIQ library released today, version **3.0.1**. If you need assistance getting the latest version or have any other issues specific to the library itself don't hesitate shoot an email to the dev team: dev@chartiq.com

Closing this issue. Thanks for reporting it!

**Comment by vanchelo** [comment]
@jaythomas Cool! Thank you!
